subsys/bluetooth/crypto/bt_crypto.c

```diff
@@ -261,18 +261,34 @@ int bt_crypto_h7(const uint8_t salt[16], const uint8_t w[16], uint8_t res[16])
 
 int bt_crypto_h8(const uint8_t k[16], const uint8_t s[16], const uint8_t key_id[4], uint8_t res[16])
 {
+	uint8_t key_id_s[4];
+	uint8_t iks[16];
+	uint8_t ks[16];
+	uint8_t ss[16];
 	int err;
-	uint8_t ik[16];
 
-	err = bt_crypto_aes_cmac(s, k, 16, ik);
+	LOG_DBG("k %s", bt_hex(k, 16));
+	LOG_DBG("s %s", bt_hex(s, 16));
+	LOG_DBG("key_id %s", bt_hex(key_id, 4));
+
+	sys_memcpy_swap(ks, k, 16);
+	sys_memcpy_swap(ss, s, 16);
```

Comment:
Why are we swapping these? 

Should they be swapped for both big and little endian?

Reply:
Have these functions been tested with a big endian device? 

The documentation for the `bt_crypto` functions does not specify anything regarding the endianess of the inputs, so it sounds like its mostly based on assumptions? 

In any case, you are being consistent with the rest of `bt_crypto.c`, so no change is required in this PR.

Reply:
hmmm... I could not find any reference/documentation in the Zephyr tree as to the endian-ness of stored keys and that Bluetooth PDU contents stored as protocol endian-ness.

For now, I will believe its assumptions in BT stack and not documenting the endian-ness of key storage.